### PR TITLE
Add HTTP request handler to e2e tests

### PR DESCRIPTION
**Pull request by mrahanjam**
# Add HTTP request handler to e2e tests

## Description
This PR adds a HTTP request handler class for the e2e tests to use.

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [X] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
N/A

## Test Plan
N/A

## Screenshots
N/A

### CODE REVIEW

File: src/e2e-test/java/io/cdap/cdap/ui/utils/HttpRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    String bodyStr = Joiner.on(",").withKeyValueSeparator("=").join(body);
+    return new ContentProvider<InputStream>() {
+      @Override
+      public InputStream getInput() {
+        return new ByteArrayInputStream(bodyStr.getBytes());
+      }
+    };
+  }
+
+  private static Multimap<String, String> castMapToMultiMap(Map<String, String> header) {
```
Comment (mrahanjam) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/HttpRequestHandler.java:66]: Added this to keep the type of request body and headers consistent. I think it avoids confusion at a higher level (see usage in `loginIfRequired` method of helper class)
https://github.com/cdapio/cdap-ui/blob/6e1f764232a9b8d1890638d58d909265a62893b1/src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java#L54

File: src/e2e-test/features/admin.feature [hunk 2]
```diff
@@ -30,6 +31,7 @@ Feature: Admin - Validate profile setting
 
   @ADMIN_TEST
   Scenario: Allow user to save valid preference at instance level after fixing error
+    Given Login if required
```
Comment (njbriggs) [inline src/e2e-test/features/admin.feature]: Is there an equivalent of `before` or `beforeEach` in Cucumber?

**Review by njbriggs — APPROVED** [review]
One question that might help make cleaner tests, otherwise LGTM

File: src/e2e-test/features/admin.feature (hunk 2 above)
Comment (mrahanjam) [reply src/e2e-test/features/admin.feature]: I think there is but I am not sure if it's for all feature files or can be tied to a single feature file (we don't call the `loginIfRequired` method in all feature files). I'll update once I have a better idea

**Review by sumengwang — APPROVED** [review]

**Review by de-lan — APPROVED** [review]
LGTM

**Pull request merged**